### PR TITLE
[Feature] List examples in Extension documentation

### PR DESCRIPTION
**Pull request by linawolf**
List all examples in the documentation and link to where they are explained in the documentation.

Squashed the commit manually.

### CODE REVIEW

File: Documentation/Index.rst [hunk 1]
```diff
@@ -52,4 +52,3 @@ Core Documentation Code Examples
    Introduction/Index
    Users/Index
    Maintainers/Index
-
```
Comment (gilbertsoft) [inline Documentation/Index.rst:55]: Imho files better end with an LF.

**Review by gilbertsoft — CHANGES_REQUESTED** [review]

File: Documentation/Index.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/Index.rst:55]: I always thought empty lines on the end of a file are evil...

Comment (linawolf) [reply Documentation/Index.rst:55]: It seems like LFs at the end of files get automatically added, if you look here: https://github.com/linawolf/TYPO3CMS-Code-Examples/edit/feature-docu/Documentation/Index.rst the LF is still there. Same is locally in my browser. For the same reason it seems to be impossible to make a second commit that would remove this change.

Comment (gilbertsoft) [reply Documentation/Index.rst:55]: Your link does not work, it's an edit link and I ofc have no write access to your fork.

Comment (linawolf) [reply Documentation/Index.rst:55]: can you see this one? it also has the empty line in line 55: https://raw.githubusercontent.com/TYPO3-Documentation/TYPO3CMS-Code-Examples/22521c4e0049eaf22af8d6266e5a9ffe3ab6e658/Documentation/Index.rst

**Pull request merged**